doc/sphinx/Pacemaker_Development/helpers.rst

```diff
@@ -4,6 +4,64 @@ C Development Helpers
 .. index::
    single: unit testing
 
+Refactoring
+###########
+
+Pacemaker uses an optional tool called `coccinelle <https://coccinelle.gitlabpages.inria.fr/website/>`_
+to do automatic refactoring.  coccinelle is a very complicated tool that can be
+difficult to understand, and the existing documentation makes it pretty tough
+to get started.  Much of the documentation is either aimed at kernel developers
+or takes the form of grammars.
+
+However, it can apply very complex transformations across an entire source tree.
+This is useful for tasks like code refactoring, changing APIs (number or type of
+arguments, etc.), catching functions that should not be called, and changing
+existing patterns.
+
+coccinelle is driven by input scripts called `semantic patches <https://coccinelle.gitlabpages.inria.fr/website/docs/index.html>`_
+written in its own language.  These scripts bear a passing resemblance to source
+code patches.  They scripts tell coccinelle how to match and modify a piece of
```

Comment:
s/They/The?

Reply:
Reworded several things here - too much "These scripts" or "The scripts" repetition.